apps/system/js/audio_channel_controller.js

```diff
@@ -0,0 +1,241 @@
+    } else if (policy.isAllowedToPlay != null) {
+      this._pause();
+    }
+    if (policy.isNeededToFadeOut) {
+      this._fadeOut();
+    } else if (policy.isNeededToFadeOut != null) {
+      this._fadeIn();
+    }
+    if (policy.isNeededToVibrate) {
+      this._vibrate();
+    }
+    callback && callback(policy);
+    this._policy = {};
+  };
+
+  /**
+   * Get active state.
+   *
+   * @return {Boolean}
+   */
+  AudioChannelController.prototype.isActive = function() {
+    return this._states.active;
+  };
+
+  /**
+   * Get state of playing.
+   *
+   * @return {Boolean}
+   */
+  AudioChannelController.prototype.isPlaying = function() {
+    return this._states.playing;
+  };
+
+  /**
+   * Get state of fading out.
+   *
+   * @return {Boolean}
+   */
+  AudioChannelController.prototype.isFadingOut = function() {
+    return this._states.fadingOut;
+  };
+
+  /**
+   * Get state of vibrating.
+   *
+   * @return {Boolean}
+   */
+  AudioChannelController.prototype.isVibrating = function() {
+    return this._states.vibrating;
+  };
+
+  /**
+   * Play the audio channel.
+   */
+  AudioChannelController.prototype._play = function() {
+    this._states.playing = true;
+    var promise = new Promise((resolve) => {
+      var request = this._channel.setMuted(false);
+      request.onsuccess = () => {
+        resolve();
+      };
+      request.onerror = () => {
+        throw 'Cannot play the audio channel.';
+      };
+    });
+    promise.then(() => {
+      this.app.debug('Play the audio channel.');
+    }).catch(e => {
+      this.app.debug(e);
+    });
+  };
+
+  /**
+   * Fade in the audio channel.
+   */
+  AudioChannelController.prototype._fadeIn = function() {
+    this._setVolume(FADE_IN_VOLUME);
+    this._states.fadingOut = false;
+  },
+
+  /**
+   * Fade out the audio channel.
+   */
+  AudioChannelController.prototype._fadeOut = function() {
+    this._setVolume(FADE_OUT_VOLUME);
+    this._states.fadingOut = true;
+    !this._states.playing && this._play();
+  },
+
+  /**
+   * Pause the audio channel.
+   */
+  AudioChannelController.prototype._pause = function() {
+    this._states.playing = false;
+    var promise = new Promise((resolve) => {
+      var request = this._channel.setMuted(true);
+      request.onsuccess = () => {
+        resolve();
+      };
+      request.onerror = () => {
+        throw 'Cannot pause the audio channel.';
+      };
+    });
+    promise.then(() => {
+      this.app.debug('Pause the audio channel');
+    }).catch(e => {
+      this.app.debug(e);
+    });
+  };
+
+  /**
+   * Vibrate for one second
+   * and the vibration pattern is [200, 100, 200, 100, ...].
+   */
+  AudioChannelController.prototype._vibrate = function() {
+    var intervalId = setInterval(() => {
+      navigator.vibrate(200);
+    }, 300);
+    this._states.vibrating = true;
+    setTimeout(() => {
+      clearInterval(intervalId);
+      this._states.vibrating = false;
+    }, VIBRATION_DURATION);
```

Comment:
This looks like you want the vibration pattern to be [200, 300, 200, 300], right? so maybe we can just write:

```JavaScript
navigator.vibrate([200, 300, 200, 300]);
```
instead of using `setInterval` and `setTimeout` to combine the vibrations, and it's total 1 second.